src/main/kotlin/se/uulm/snowballr/backend/repository/association/ReadingListTableRepo.kt

```diff
@@ -0,0 +1,100 @@
+package se.uulm.snowballr.backend.repository.association
+
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.eq
+import org.jetbrains.exposed.sql.and
+import org.jetbrains.exposed.sql.count
+import org.jetbrains.exposed.sql.deleteWhere
+import org.jetbrains.exposed.sql.insert
+import org.jetbrains.exposed.sql.selectAll
+import se.uulm.snowballr.backend.db.IDatabase
+import se.uulm.snowballr.backend.model.EntityType
+import se.uulm.snowballr.backend.model.SnowballRException.NotFoundException
+import se.uulm.snowballr.backend.table.PaperTable
+import se.uulm.snowballr.backend.table.UserTable
+import se.uulm.snowballr.backend.table.association.ReadingListTable
+import java.util.UUID
+
+/**
+ * Defines an interface for repository operations related to the [ReadingListTable].
+ *
+ * This interface provides an abstraction for handling persistence and retrieval operations for reading list entries.
+ * By using this interface, the functionality for managing reading lists remains decoupled from the specifics of the
+ * database layer, promoting flexibility and maintainability in the codebase.
+ */
+interface IReadingListTableRepo {
+    /**
+     * Add the provided paper to the reading list of the provided user.
+     */
+    suspend fun createReadingListEntry(userId: UUID, paperId: UUID)
+
+    /**
+     * Remove the provided paper from the reading list of the provided user.
+     */
+    suspend fun removeReadingListEntry(userId: UUID, paperId: UUID)
+
+    /**
+     * @returns Wether the provided paper is on the reading list of the provided user.
+     */
+    suspend fun isPaperOnReadingList(userId: UUID, paperId: UUID): Boolean
+
+    /**
+     * @returns The [List] of all ids of the papers the provided user has on their reading list.
+     */
+    suspend fun getAllReadingListEntries(userId: UUID): List<UUID>
+}
+
+class ReadingListTableRepo(
+    private val db: IDatabase,
+) : IReadingListTableRepo {
+    private fun throwIfUserDoesNotExist(userId: UUID) {
+        if (UserTable.selectAll().where { UserTable.id eq userId }.count() == 0L) {
+            throw NotFoundException(EntityType.USER, userId.toString())
+        }
+    }
+
+    private fun throwIfPaperDoesNotExist(paperId: UUID) {
+        if (PaperTable.selectAll().where { PaperTable.id eq paperId }.count() == 0L) {
+            throw NotFoundException(EntityType.PAPER, paperId.toString())
+        }
+    }
+
+    override suspend fun createReadingListEntry(userId: UUID, paperId: UUID) = db.dbQuery {
+        throwIfUserDoesNotExist(userId)
+        throwIfPaperDoesNotExist(paperId)
+
+        if (ReadingListTable.selectAll().where {
+                (ReadingListTable.userId eq userId) and (ReadingListTable.paperId eq paperId)
+            }.count() > 0
+        ) {
```

Comment:
Here we can reuse `isPaperOnReadingList`
```suggestion
        if (isPaperOnReadingList(userId, paperId)) {
```

Reply:
AFAIK nested transactions are ok, adding this to improve readability.